metric_name_builder.go

```diff
@@ -81,29 +80,47 @@ var perUnitMap = map[string]string{
 	"y":  "year",
 }
 
-// BuildCompliantMetricName builds a Prometheus-compliant metric name for the specified metric.
-//
-// Metric name is prefixed with specified namespace and underscore (if any).
-// Namespace is not cleaned up. Make sure specified namespace follows Prometheus
-// naming convention.
+// MetricNamer is a helper struct to build metric names.
+type MetricNamer struct {
+	Namespace          string
+	WithMetricSuffixes bool
+	UTF8Allowed        bool
+}
+
+// Metric is a helper struct that holds information about a metric.
+type Metric struct {
+	Name string
+	Unit string
+	Type MetricType
+}
+
+// Build builds a metric name for the specified metric.
 //
+// If UTF8Allowed is true, the metric name is returned as is, only with the addition of type/unit suffixes and namespace preffix if required.
+// Otherwise the metric name is normalized to be Prometheus-compliant.
 // See rules at https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels,
 // https://prometheus.io/docs/practices/naming/#metric-and-label-naming
-// and https://github.com/open-telemetry/opentelemetry-specification/blob/v1.38.0/specification/compatibility/prometheus_and_openmetrics.md#otlp-metric-points-to-prometheus.
-func BuildCompliantMetricName(metric pmetric.Metric, namespace string, addMetricSuffixes bool) string {
+func (mn *MetricNamer) Build(metric Metric) string {
```

Comment:
What if `metric.Type == MetricTypeUnknown`? Should this panic as an illegal input? Is there a test case for it?

Reply:
In Prometheus, at least for now, it is possible to ingest metrics with unknown types without errors, so I wouldn't panic and not even return errors actually. 

Since there is nothing in the spec saying that a metric of unknown type should receive a prefix/suffix, I believe we're following the rules and just doing nothing

Reply:
Just make sure that it is covered iby tests and resolve this conversation.